### DISCUSSION TITLE
StringData structure vary on using workflows with "UsableAsActivity"

### DISCUSSION THREAD

**Discussion by sagargolu** [elsa-workflows/elsa-core#6231] [Q&A]
@sfmskywalker ,
I've noticed that when saving or publishing the main workflow, if it includes sub-workflows with 'UsableAsActivity' set to true, the StringData column sometimes stores all the activities from the sub-workflows used in it, while other times it remains empty. Could you explain why this happens?

**Comment by sfmskywalker** [comment]
That’s not what’s supposed to happen, so it sounds like a bug. I’ll see if I can reproduce this. Thanks for reporting!

**Reply by sagargolu** [reply]
Here, I can tell you the behavior of the flow where that is happening.

1. Create a sub-workflow, add some activities to it, and publish it.
2. Create a main workflow, include the sub-workflow in it, and publish the main workflow. This behaves as expected.
3. Revisit the same sub-workflow, make some changes, and publish it again.
4. Go back to the main workflow and attempt to upgrade the sub-workflow version. At this stage, the behavior changes: the main workflow stores all the root activities of the sub-workflow in the StringData field.
5. However, if the sub-workflow is removed from the main workflow and then re-added, the behavior returns to normal, with proper storage of StringData.

**Reply by sagargolu** [reply]
@sfmskywalker 

Did you find anything on this? I did some patching for now to overcome from the issue, I am removing "root" object at the time of saving if activity is sub-workflow.
